### ISSUE TITLE
Downloaded data differs from web data?

### ISSUE THREAD

**Issue by frasator** [obophenotype/human-phenotype-ontology#5770]
**Bug description**

I'm using the following files:
_genes_to_phenotype.txt_ ( let's call it  **g2p**)
_phenotype_to_genes.txt_ (let's call it **p2g**)
_phenotype.hpoa_ (let's call it hpoa)

The last column of **g2p** or **p2g** contains the **disease ID for link** as shown in the format line.
For example if I grep those files with this disease **OMIM:609535** id,  I found this lines repectively:
**g2p**
`1557    CYP2C19 HP:0000007      Autosomal recessive inheritance                 -       mim2gene        OMIM:609535`
**p2g**
`HP:0000007      Autosomal recessive inheritance 1557    CYP2C19 -       mim2gene        OMIM:609535`

This seems ok for now, then I would like to get the name and more related information with this **OMIM:609535**, and when I grep **hpoa** with the id, no entry is found :(

So, if it appears in **g2p** and **p2g** why is not in **hpoa**? could be a bug? what am I missing?

Then went to [https://hpo.jax.org/ ](https://hpo.jax.org/ ) to check if happens the same,  and **OMIM:609535** is not found.

Then, tried to search **HP:0000007** but, neither the gene **CYP2C19** and the disease **OMIM:609535** appears as associated with the **HP:0000007** term.

Also, tried to search the gene **CYP2C19** but has no HPO or Disease associations.

The following **disease ID for link** found in **g2p** and **p2g** not appears in the **hpoa**.
Use this command to get the list:
`comm -23  <(cat phenotype_to_genes.txt | grep -v "#" | cut -f7 | sort -u ) <(cat phenotype.hpoa | cut -f1 | sort -u )`
```
OMIM:110800
OMIM:111150
OMIM:111400
OMIM:245300
OMIM:300087
OMIM:300910
OMIM:602025
OMIM:605429
OMIM:608622
OMIM:609338
OMIM:609535
OMIM:612460
OMIM:614740
OMIM:615457
OMIM:615881
OMIM:616903
ORPHA:64740
```
None of the diseases ids on this list appears on the web [https://hpo.jax.org/ ](https://hpo.jax.org/ ) in the moment i'm writing this.

Here some commads to filter those diseases lines that appears in  **g2p** and **p2g** and not appear in **hpoa**.

`cat genes_to_phenotype.txt | egrep -w "OMIM:110800|OMIM:111150|OMIM:111400|OMIM:245300|OMIM:300087|OMIM:300910|OMIM:602025|OMIM:605429|OMIM:608622|OMIM:609338|OMIM:609535|OMIM:612460|OMIM:614740|OMIM:615457|OMIM:615881|OMIM:616903|ORPHA:64740"`

`cat phenotype_to_genes.txt | egrep -w "OMIM:110800|OMIM:111150|OMIM:111400|OMIM:245300|OMIM:300087|OMIM:300910|OMIM:602025|OMIM:605429|OMIM:608622|OMIM:609338|OMIM:609535|OMIM:612460|OMIM:614740|OMIM:615457|OMIM:615881|OMIM:616903|ORPHA:64740"`

`cat phenotype.hpoa | egrep -w "OMIM:110800|OMIM:111150|OMIM:111400|OMIM:245300|OMIM:300087|OMIM:300910|OMIM:602025|OMIM:605429|OMIM:608622|OMIM:609338|OMIM:609535|OMIM:612460|OMIM:614740|OMIM:615457|OMIM:615881|OMIM:616903|ORPHA:64740"`
_This command will return nothing._

 
 
 

Thanks in advance

**Comment by pnrobinson** [comment]
OMIM:609535 is not a disease, it is a pharmacogenomic trait. At present, such traits are not represented in the phenotype.hpoa file, which is just for diseases.
Could we ask for your feedback about the best way for the HPO project to represent pharmacogenetic traits?

**Comment by frasator** [comment]
Thank you very much for your answer, expected to find only diseases on the files like on the web.
I have been working with HPO project data for only two weeks and at this moment I am just validating my scripts that uses downloaded files against the web to test my results.
I'm not sure about how to represent pharmacogenomic traits, but if the web only appears **genes**, **HPO terms** and **diseases**, and in the files **g2p** and **p2g**, under the column called **disease ID for link**  there are ids that are not diseases, it is a little messy.


So maybe, add a new colum indicating the type (disease, pharmacogenomic trait...) or put those lines in a new file with the same format but changing the **disease ID for link** for **pharmacogenomic traits ID for link**.
The same for the **hpoa** file, could be one called **disease-annotation.hpoa** and other for **pharmagenomic-trait-annotation.hpoa**. The same could be applied if there exists other types. 
Or maybe everthing that is not disease all together in a **non-disease-annotation.hpoa**.
Any representation consistent and reproducible is fine.

What about ? [ORPHA:64740](https://www.orpha.net/consor/cgi-bin/OC_Exp.php?Expert=64740&lng=EN)
It says is **Recurrent acute pancreatitis** and is missing on the web and in the **hpoa** file, but, appears in the  **g2p** and **p2g** files, where is the bug, on the web or in the files?.

Also there is a typo in the **g2p** and **p2g** files, the column names (**HPO-Term-Name**, **HPO-Term-ID**) and (**HPO-id** and **HPO label**) in the **#Format** are different but represent the same data, and why **HPO label** has a space instead of hypen like the others (some columns has hyphens and others spaces, would be nice all hyphen or all spaces).
**p2g**(HPO-id) represents same data as **g2p**(HPO-Term-Name)
**p2g**(HPO label) represents same data as **g2p**(HPO-Term-ID)


 In the **g2p** file seems **HPO-Term-Name** and **HPO-Term-ID** are swapped
**g2p** 
HPO-Term-Name     i.e. **_HP:0004322_** should be **_Short stature_**
HPO-Term-ID      i.e. **_Short stature_** should be  **_HP:0004322_**

**Comment by pnrobinson** [comment]
@drseb Did you see the format error in the g2p file?
@frasator Thanks for this input. We are aware of the issue and the idea of non-disease-annotation.hpoa or something like that is good. At the moment, we are strapped for resources and I do not think we will be able to implement this in the very near future. I am going to leave this issue open, and depending on the grant funding I hope we will be able to move forward.

**Comment by frasator** [comment]
Thank you very much

**Comment by drseb** [comment]
Hi @frasator. The basis for the g2p and p2g files are not the hpoa file, it is [this file](http://purl.obolibrary.org/obo/hp/hpoa/phenotype_annotation.tab).

**Comment by pnrobinson** [comment]
I think we can close this now, please reopen if there are new problems.
